### PR TITLE
Add a regression test for issue #4888

### PR DESCRIPTION
**Pull request by mattpap**
Performance of many (`N=50`) webgl glyphs is still not great, compared to canvas and svg:
image

The number for webgl is still quite generous, as it can vary a bit, up to 900~1000 ms in this test.

Improves `Plot.add_tool(tool_alias)` as well, which I used for testing; bokehjs already allows this.

### CODE REVIEW

**Comment by ianthomas23** [comment]
> Performance of many (N=50) webgl glyphs is still not great, compared to canvas and svg

This is an inappropriately small example to pass any form of judgement on WebGL performance. https://docs.bokeh.org/en/latest/docs/user_guide/output/webgl.html#when-to-use-webgl states that you should use WebGL if you have tens of thousands of points, here there are just 500. This is the rendering equivalent of getting in a sports car to travel a distance of ten metres and then complaining that you could have walked it faster.

**Comment by mattpap** [comment]
Sure, the default for webgl is large data, however I hoped that the gap in setup time between webgl and other backends won't be as big as this. This is precisely why tests like this are needed, so that we know and set our expectations right. Of course it would be even better if setup time was actually measured in the test, but this will have to wait some more.

**Comment by mattpap** [comment]
btw. `N` is the number of glyphs, so the data is `N*10` in this test, which is still very few data points for webgl.

**Comment by mattpap** [comment]
> (...) when that is never going to be case for problems of this size within the constraints of how Bokeh rendering occurs.

I'm not sure if this is true actually. Our webgl backend is optimized for large data, but there is another approach. I'm not sure if regl does this, but there are libraries that provide a canvas like API for rendering with webgl (something similar to what we do for svg backend). Then a hybrid approach would possibly allow to render efficiently across the data size spectrum. In fact, at some point there was an idea that we should switch to webgl rendering only (perhaps before svg backend was a thing).

**Comment by bryevdv** [comment]
FYI although this case may highlight webgl, this situation is not in any way confined to webgl. This example, with 56 glyphs (28 scatter and 28 text) and *much* less data loads so slowly on canvas a few times recently I wondered if it was broken/non-functional:

https://docs.bokeh.org/en/3.0.1/docs/examples/basic/scatters/markers.html

Best practice with Bokeh has *long* been "bigger data sources, fewer glyphs" and to be clear, that is *fine*. There are always trade-offs tool can excel in every use-case. What is important is that we provide users with clear information and guidance about trade-offs. Which is not to say that if someone has the time and interest they can't or shouldn't *investigate* what and how any given thing might be improved. But I agree with @ianthomas23 that the tracker is public and searchable and speculation or editorializing here does not help users or us, and that paying more attention to where different kinds of discussions occur would be a good thing. 

IMO, all that needed to be said here:

> This PR adds a test that reports performance for different drawing backend in the case of a high (N=~50) number of line glyphs. The results in on my local machine are:
>
>  | canvas | 171 ms |
>  | svg     | 117 ms |
> | webgl | 716 ms |
> 
> We could use this to refine our guidance about the best way to use Bokeh in some situations (I've opened a seprate  <discussion or issue> for that)

**Comment by mattpap** [comment]
> (...) loads so slowly [1] on canvas (...)
> https://docs.bokeh.org/en/3.0.1/docs/examples/basic/scatters/markers.html
> p = figure(title="Bokeh Markers", toolbar_location=None, output_backend="webgl")

This is a webgl example. On canvas it loads in a fraction of the time. Still could a bit faster though.

**Comment by bryevdv** [comment]
Huh, not sure when that happened. I still stand by the observation, as I have had to advise users to consolidate many glyphs on more occasions that I can count over the years, and also by the stance that it is OK for different things to be good for different use cases, as long as we are clear, e.g. "webgl is most usefule and appropriate when there are hundreds of thousands or more points".

File: src/bokeh/models/plots.py [hunk 1]
```diff
@@ -312,10 +312,14 @@ def add_tools(self, *tools: Tool) -> None:
...
-
-            self.toolbar.tools.append(tool)
+            if isinstance(tool, str):
+                tool_obj = Tool.from_string(tool)
+            elif isinstance(tool, Tool):
+                tool_obj = tool
+            else:
+                raise ValueError(f"expected a string or Tool instance, got {tool!r}")
+
+            self.toolbar.tools.append(tool_obj)
```
Comment (bryevdv) [inline src/bokeh/models/plots.py:322]: All of this and related tests seem unrelated, and therefore better in a dedicated PR with an issue to document an expanded API for the changelog
